### PR TITLE
NO-ISSUE: BPMN Editor: Add new command to select all nodes in diagram

### PR DESCRIPTION
**Pull request by EOlegovich**
Added command to select full process

### CODE REVIEW

**Comment by tiagobento** [comment]
@EOlegovich Hi! Thanks a lot for the contribution! I took the liberty to edit the PR title, and assigned @inodeman as reviewer. 

Also, I think the Keyboard Shortcut name should change to "Select all", WDYT?

**Review by inodeman — CHANGES_REQUESTED** [review]
Hi @EOlegovich great job! All I can think of is write the current year in copyright for new files and write unit tests for the changes.

File: packages/stunner-editors/kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/session/command/impl/FullSelectionSessionCommand.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
```
Comment (jomarko) [inline packages/stunner-editors/kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/session/command/impl/FullSelectionSessionCommand.java]: I think this is not appropriate year

File: packages/stunner-editors/kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/session/command/impl/FullSelectionSessionCommandExecutedEvent.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
```
Comment (jomarko) [inline packages/stunner-editors/kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/session/command/impl/FullSelectionSessionCommandExecutedEvent.java]: similar comment as above

File: packages/stunner-editors/kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/session/command/impl/FullSelectionSessionCommand.java [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        }
+    }
+
+    @Override
+    public <V> void execute(final Callback<V> callback) {
+        if (getSession() != null && null != getSession().getSelectionControl()) {
+            try {
+                final SelectionControl<AbstractCanvasHandler, Element> selectionControl = getSession().getSelectionControl();
+                for(String item : selectionControl.getItems().keySet()) {
+                    getSession().getSelectionControl().select(item);
```
Comment (jomarko) [inline packages/stunner-editors/kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/session/command/impl/FullSelectionSessionCommand.java]:
```suggestion
                    selectionControl.select(item);
```

File: packages/stunner-editors/kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/session/command/impl/FullSelectionSessionCommand.java [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    @Override
+    public boolean accepts(final ClientSession session) {
+        return session instanceof EditorSession;
+    }
+
+    protected void onKeyDownEvent(final KeyboardEvent.Key... keys) {
+        handleCtrlA(keys);
+    }
+
+    private void handleCtrlA(final KeyboardEvent.Key[] keys) {
```
Comment (jomarko) [inline packages/stunner-editors/kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/session/command/impl/FullSelectionSessionCommand.java:83]: as this private method is used just single time, I think we do not need to introduce it and put the code directly into `onKeyDownEvent`

**Review by jomarko — COMMENTED** [review]
Thank you for this contribution.  May I ask you of you tried effect of this changes for:
- BPMN Swimmlanes and Subprocesses
- DMN Decision Service nodes

Generally for nodes that are just containers of other nodes.

**Comment by inodeman** [comment]
Hi @EOlegovich thanks for the copyright correction, I think with just one Unit test file for the main changes will be enough, thanks

**Comment by tiagobento** [comment]
@EOlegovich Any updates? Please take a look at @inodeman's comment above.

**Comment by EOlegovich** [comment]
Hello.
Sorry for the long delay in my reply.
Unfortunately, I don't have the opportunity to write a test.

**Pull request closed without merging**